functions/go/apply-replacements/run_js.go

```diff
@@ -37,19 +41,40 @@ func applyReplacements(input []byte) ([]byte, error) {
 	return fn.Run(fn.ResourceListProcessorFunc(replacements.ApplyReplacements), []byte(input))
 }
 
-func resourceListProcessorWrapper() js.Func {
-	// TODO: figure out a better way to surface a golang error to JS environment.
-	// Currently error is surfaced as a string.
-	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+func resourceListProcessorWrapper(resourceList *[]byte) js.Func {
+	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) any {
 		if len(args) != 1 {
 			return "Invalid number of arguments passed"
```

Comment:
Perhaps this entire function could simply either return the resourceList or empty string, always deferring to the other function to get error messages, rather than mixing those approaches. Would require an update to https://github.com/GoogleContainerTools/kpt/pull/3830 if we take that route.

Reply:
That's a good point. I think deferring to the other function to get errors makes a lot of sense. Would like to hear @droot @mortent opinion on this.